Block/Product/Review.php

```diff
@@ -50,17 +51,35 @@ public function __construct(
      */
     public function getCollectionSize()
     {
-        $collection = $this->_reviewsColFactory->create()->addStoreFilter(
-            $this->_storeManager->getStore()->getId()
-        )->addStatusFilter(ModelReview::STATUS_APPROVED)->addEntityFilter(
-            'product',
-            $this->getProductId()
-        );
+        $product = $this->getProduct();
+        $collection = $this->_reviewsColFactory->create()
+            ->addStoreFilter($this->_storeManager->getStore()->getId())
+            ->addStatusFilter(ModelReview::STATUS_APPROVED)
+        ;
 
         if (!$this->helper->isProductsOpinionsEnabled()) {
             $collection->addFieldToFilter('title', array('neq' => Data::OPINION_TITLE));
+            $collection->addEntityFilter('product', $product->getId());
```

Comment:
czy getProduct może zwrócić null? jeśli tak to tutaj i w dwóch miejscach poniżej trzeba to obsłużyć

Reply:
Dzięki, słuszna uwaga. Poprawione.